conftest.py

```diff
@@ -0,0 +1,21 @@
+import pytest
+from users.models import User
+from knox.auth import AuthToken
+from rest_framework.test import APIClient
+
+@pytest.fixture
```

Comment:
it is better to do it that way because we use the user fixture, we don't want to create a user in the auth_client fixture
```python
@pytest.fixture
def user():
    user = User.objects.create_user('ahmed', 'ahmed@email.com', 'Ahmed_1234')
    return user
```
```python
@pytest.fixture
def get_client(user):
    def api_client(user_instance=None):
        if user_instance is None:
            client = APIClient()

            login_response = client.post('/login/', dict(username="ahmed", password="Ahmed_1234"))

            token = login_response.data["token"]

            client.credentials(HTTP_AUTHORIZATION='Token ' + token)

            return client, user.id
        else:
            client = APIClient()
            
            random_user = User.objects.create_user(user_instance['username'], user_instance['email'], user_instance['password'])

            login_response = client.post('/login/', dict(username=user_instance["username"], password=user_instance["password"]))

            token = login_response.data["token"]

            client.credentials(HTTP_AUTHORIZATION='Token ' + token)

            return client, random_user.id
    
    return api_client
```